sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/azure_configuration_client_async.py

```diff
@@ -209,15 +209,18 @@ def list_configuration_settings(
         if match_condition == MatchConditions.IfMissing:
             error_map[412] = ResourceExistsError
 
-        key_value = await self._impl.get_key_value(
-            key=key,
-            label=label,
-            if_match=prep_if_match(etag, match_condition),
-            if_none_match=prep_if_none_match(etag, match_condition),
-            error_map=error_map,
-            **kwargs
-        )
-        return ConfigurationSetting._from_key_value(key_value)
+        try:
+            key_value = await self._impl.get_key_value(
+                key=key,
+                label=label,
+                if_match=prep_if_match(etag, match_condition),
+                if_none_match=prep_if_none_match(etag, match_condition),
+                error_map=error_map,
+                **kwargs
+            )
+            return ConfigurationSetting._from_key_value(key_value)
+        except ResourceNotModifiedError:
```

Comment:
Is 304 not listed in the swagger as a success response? If it is, can't we just not add it to the error map instead of try/except:ing?